website/project/model.py

```diff
@@ -2239,6 +2239,11 @@ def add_contributor(self, contributor, permissions=None, visible=True,
 
             contributor_added.send(self, contributor=contributor, auth=auth)
             return True
+
+        #Permissions must be overridden if changed when contributor is added to parent he/she is already on a child of.
+        elif contrib_to_add in self.contributors and permissions is not None:
+            self.set_permissions(contrib_to_add, permissions, True)
```

Comment:
This function has an optional `save` parameter with `False` as the default option so we do not want the default here to be that it always saves. Instead, if save is true, then save the node object. For example: https://github.com/CenterForOpenScience/osf.io/blob/develop/website/project/model.py#L2237-#L2238

Sorry if I wasn't clear earlier!